data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/DefaultAcknowledgementSetManager.java

```diff
@@ -36,12 +39,23 @@ public DefaultAcknowledgementSetManager(
     public DefaultAcknowledgementSetManager(final ExecutorService callbackExecutor, final Duration waitTime) {
         this.acknowledgementSetMonitor = new AcknowledgementSetMonitor();
         this.executor = Objects.requireNonNull(callbackExecutor);
+        // Single thread executor should be sufficient in most cases (esp if the underlying host has 1 or 2 cores)
+        this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
The current executor is dependency injected. I think we should continue that pattern. It comes from:

https://github.com/opensearch-project/data-prepper/blob/068c57172c0dfb0c4e43d1d796e8a59fa11fb1df/data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/AcknowledgementAppConfig.java#L25-L28

Also, can we share a single executor?

Change this:

https://github.com/opensearch-project/data-prepper/blob/068c57172c0dfb0c4e43d1d796e8a59fa11fb1df/data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/AcknowledgementAppConfig.java#L27C28-L27C28

To this:

```
return Executors.newScheduledThreadPool(MAX_THREADS, callbackTheadFactory);
```

And change the interfaces to use `ScheduledExecutorService` throughout.